### ISSUE TITLE
Deployment Object Incompatible with NodeShift

### ISSUE THREAD

**Issue by RachitBhargava99** [nodeshift/nodeshift#476]
I have a use-case where I need to use NodeShift service while providing a Deployment object. Somehow, the service assumes the file to be of kind DeploymentConfig even though its kind is mentioned to be Deployment. Here is a copy of error message I get when I deploy using NodeShift:

`ERROR DeploymentConfig in version "v1beta1" cannot be handled as a DeploymentConfig: no kind "DeploymentConfig" is registered for version "extensions/v1beta1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"
ERROR Status code 400`

**Comment by lholmquist** [comment]
There is some history behind this why we decided to convert all deployments into deployment configs.  which i can't really remember :).

I think whatever reason that was, doesn't make sense to do anymore. This is something i've been wanting to change, actually, and this is probably a good opportunity now.

So for your use case, you have a deployment yaml file in your .nodeshift directory?  with the version and type defined?   


This most likely will be a semver major change since it is changing some default behavior.

**Comment by lholmquist** [comment]
There are probably a couple things that need to be done to make this change.

1. If someone is using and yaml that are put in the `.nodeshift/` directoy, we need to extract the version(if any, since it can be left blank) since we will need to use this to construct the REST call.  currently, we are defaulting to `v1` for everything

2.  Don't convert Deployments to Deployment Configs

**Comment by lholmquist** [comment]
@RachitBhargava99 I also just want to make sure you are using Nodeshift to deploy something to Openshift and not plain kubernetes?

**Comment by RachitBhargava99** [comment]
@lholmquist yes, that's right.

**Comment by lholmquist** [comment]
@RachitBhargava99 I've just released 7.1.0 which adds support for Deployments.  you just need to use the `--useDeployment` flag

**Comment by RachitBhargava99** [comment]
The flag does not seem to work as intended. I get the following error when using it.

```
ERROR Deployment.apps "<app-name>" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"thinktank", "chart":"ibm-cem", "component":"thinktank", "deploymentconfig":"<app-name>", "heritage":"Tiller", "origin":"helm-cem", "project":"<app-name>", "provider":"nodeshift", "release":"rachit", "version":"1.0.0"}: `selector` does not match template `labels`
ERROR Status code 422
```
